### PR TITLE
Fix 18 medium severity issues from code review (#58-#75)

### PR DESCRIPTION
**Pull request by chrisfromthelc**
## Summary

Addresses all 18 medium priority findings from the comprehensive code review (GitHub issues #58–#75). These fixes improve input validation, error handling, SQL efficiency, and edge-case coverage across collectors, fixers, and infrastructure classes.

### Changes by category

**Security & Input Validation**
- **#60** CLI command: add `manage_options` capability check before fixer execution
- **#64** Settings: sanitize and validate all input with proper `sanitize_callback` and `validate_callback` functions

**Error Handling & Resilience**
- **#59** AI context generator: add 15s timeout and retry with exponential backoff for HTTP requests
- **#63** GitHub updater: validate API response structure before accessing properties
- **#65** SSE log controller: add `Content-Type: text/event-stream` header and `Connection: close`
- **#70** Network connectivity: detect DNS resolution failures separately from HTTP errors
- **#73** Server environment: handle `convert_to_bytes('-1')` edge case (unlimited)
- **#75** Cron repair: handle `WP_Error` from `wp_unschedule_event()`

**SQL & Performance**
- **#67** Custom content types: add `LIMIT 100` to taxonomy term count query
- **#69** Media uploads: cap directory size walk at 50,000 files to prevent runaway I/O
- **#72** Post type counts: add `LIMIT` clause and use `COUNT(1)` for efficiency

**Code Quality & Correctness**
- **#58** Add `.phpunit.result.cache` to `.gitignore`
- **#61** Fix_Result: add strict `array` type to `$data` property
- **#62** Fixer registry: prevent duplicate fixer registration
- **#66** Block editor: add fallback for missing block type `source` property
- **#68** Filesystem permissions: detect and report `open_basedir` restrictions
- **#71** Performance: detect persistent object cache drop-in (`object-cache.php`)
- **#74** Update health: remove redundant `is_wp_error()` check
- AI formatter: refactor with `format_header()`/`format_section()` helper methods

## Test plan

- [ ] PHPCS passes on all 4 PHP version targets (8.1–8.4)
- [ ] PHPStan passes with no new errors
- [ ] PHPUnit test suite passes on all 4 PHP version targets
- [ ] Rector dry-run shows no regressions
- [ ] Verify Settings page saves and loads correctly after validation changes
- [ ] Verify CLI `wp system-report fix` requires proper capabilities
- [ ] Verify AI context generation handles timeout/retry gracefully

## Summary by Sourcery

Address medium-priority review findings by tightening validation and diagnostics across settings, collectors, formatters, and infrastructure while improving consistency with platform requirements.

New Features:
- Add in-memory caching for plugin settings reads to reduce repeated option lookups during a request.
- Report world-writable WordPress root directory permissions as a critical filesystem health issue.
- Expose more consistent health scoring and grading in the AI formatter by aligning it with the central Health_Score logic.
- Cache block registry and options row counts within collectors to avoid redundant expensive lookups.

Bug Fixes:
- Ensure CLI formatted output reports unsupported formats using translatable error messages and keep MCP handling separate from other formatters.
- Prevent duplicate fixer registrations by emitting a _doing_it_wrong() notice when an existing ID is reused.
- Adjust post type counts to use wp_count_posts with status filtering, omitting types with zero content and covering non-standard statuses.
- Handle world-writable root permissions and open_basedir/stat edge cases more safely in filesystem permissions collection.
- Fix Settings static cache behavior and tests so updates and deletes consistently invalidate cached option data.
- Align GitHub updater metadata with the plugin’s actual minimum PHP requirement of 8.1 and update tests accordingly.
- Ensure cron repair rescheduling logic reuses a cached cron array instead of reloading it on every loop iteration.
- Relax SSL certificate expiry warning thresholds and clarify why peer verification is disabled for diagnostic certificate checks.
- Update server environment PHP version health threshold to warn only when running below the plugin’s minimum supported version.
- Make SSE log streaming more robust on hosts where set_time_limit is disabled.
- Ensure the AI formatter executive summary health score matches the central scoring model and update tests to reflect new grading.
- Clarify that certain SQL diagnostics using COUNT(1) do not need LIMIT clauses and rely on indexed joins for scalability.

Enhancements:
- Replace raw SQL for post type counting with WordPress APIs and add targeted tests for empty post-type scenarios.
- Reuse counted options rows in performance collectors to drive persistent cache recommendations without duplicate queries.
- Use prepared statements with table-name placeholders in options-table queries for consistency and safety.
- Stabilize AI context generator field matching by relying on export_label keys instead of translated labels, and document coupling to collector export labels.
- Improve custom content types collector by caching global arrays locally before use and hardening against non-array globals.
- Tighten tests for filesystem permissions to account for the new world-writable root field and its non-path debug payload.
- Simplify update health failed-update reporting by removing redundant auto-update history handling.
- Clarify media uploads orphaned-attachment query behavior and document why COUNT(1) does not require a LIMIT.

### CODE REVIEW

**Comment by chrisfromthelc** [comment]
@sourcery-ai review

File: includes/formatters/class-ai-formatter.php [hunk 1]
```diff
@@ -147,32 +147,62 @@ private function build_header(): string {
...
+		/*
+		 * Delegate health score calculation to Health_Score so that the
+		 * executive summary always reflects the same score as the REST API
+		 * and admin dashboard. Previously this method used a simplified
+		 * penalty-based formula (100 - 10*criticals - 5*warnings) that could
+		 * diverge significantly from Health_Score's weighted-average approach.
+		 *
+		 * Health_Score::calculate() requires a Report_Generator instance, so we
+		 * replicate its field-level scoring logic here directly from $report_data
+		 * (good = 100 pts, warning = 40 pts, critical = 0 pts, info = excluded)
```
Comment (SourceryAI) [inline includes/formatters/class-ai-formatter.php]: **suggestion:** The health-score algorithm is now partially duplicated here and could drift from Health_Score’s central logic.

This method now re-implements field-level scoring (good=100, warning=40, critical=0, info excluded) and only delegates to `Health_Score::score_to_grade()` for the label. Any future change to scoring weights or inclusion rules in `Health_Score` would have to be manually mirrored here to keep the executive summary consistent with the REST/admin views.

To avoid this duplication, consider either:
- Adding a helper in `Health_Score` that takes `$report_data` and returns both numeric score and grade, or
- Centralizing the per-field weighting constants in `Health_Score` and referencing them here.

That way, score logic remains defined in one place and stays in sync across all surfaces.
